### PR TITLE
[`BLIP`] fix doctest

### PR DESCRIPTION
**Pull request by younesbelkada**
# What does this PR do?

This PR fixes `BLIP` doctest. 

Link to failing job: https://github.com/huggingface/transformers/actions/runs/3964164193 

The docstring of the `forward` method of `BlipForQuestionAnswering` has been corrected to educate users on how to correctly use this module after https://github.com/huggingface/transformers/pull/21021 being merged. 
The logic now of the `forward` method is pretty much similar to [`T5`](https://github.com/huggingface/transformers/blob/main/src/transformers/models/t5/modeling_t5.py#L1592-L1610).  

cc @ydshieh 💯

### CODE REVIEW

File: src/transformers/models/blip/modeling_blip.py [hunk 1]
```diff
@@ -1176,17 +1176,28 @@ def forward(
...
+        >>> outputs = model.generate(**inputs)
+        >>> print(processor.decode(outputs[0], skip_special_tokens=True))
+        2
         ```"""
         if labels is None and decoder_input_ids is None:
             raise ValueError(
                 "Either `decoder_input_ids` or `labels` should be passed when calling `forward` with"
                 " `BlipForQuestionAnswering`. if you are training the model make sure that `labels` is passed, if you"
-                " are using the model for inference make sure that `decoder_input_ids` is passed."
+                " are using the model for inference make sure that `decoder_input_ids` is passed or call `generate`"
```
Comment (ydshieh) [inline src/transformers/models/blip/modeling_blip.py:1200]: No for the change in this PR, but `if you are training` should be `If you are training`. And the second part should be `, and if you ...` or `. If you ...`

**Review by ydshieh — APPROVED** [review]
LGTM 💯 with -1 --> I want to have ♾️ cats instead of just 2 😹

File: src/transformers/models/blip/modeling_blip.py (hunk 1 above)
Comment (ydshieh) [reply src/transformers/models/blip/modeling_blip.py:1200]: Let @NielsRogge for a second eye 👀 on this PR.

File: src/transformers/models/blip/modeling_blip.py [hunk 2]
```diff
@@ -1176,17 +1176,28 @@ def forward(
...
-        >>> text = "How many cats are in the picture?"
 
+        >>> # training
+        >>> text = "How many cats are in the picture?"
+        >>> label = "2"
         >>> inputs = processor(images=image, text=text, return_tensors="pt")
+        >>> labels = processor(text=label, return_tensors="pt").input_ids
 
+        >>> inputs["labels"] = labels
         >>> outputs = model(**inputs)
```
Comment (NielsRogge) [inline src/transformers/models/blip/modeling_blip.py:1187]:
```suggestion
        >>> inputs["labels"] = labels
        >>> outputs = model(**inputs)
        >>> loss = outputs.loss
        >>> loss.backward()
```

File: src/transformers/models/blip/modeling_blip.py (hunk 1 above)
Comment (NielsRogge) [reply src/transformers/models/blip/modeling_blip.py:1200]: Yes agree here!

**Review by NielsRogge — APPROVED** [review]
Thanks for fixing!
